network-api/networkapi/wagtailpages/pagemodels/libraries/research_hub/detail_page.py

```diff
@@ -204,15 +206,19 @@ def __str__(self) -> str:
     def clean(self) -> None:
         super().clean()
 
+        url_set = bool(self.url)
+        page_set = bool(self.page)
+        document_set = bool(self.document)
```

Comment:
What do you think of phrasing these variables in a question style? E.g. `is_url_set`. `url_set` makes it sound like a set of urls (multiple).

Reply:
Ah, here it is. I this page / document / external link pattern is extremely common. It's not even limited to library pages. What do you think about moving it into a util mixin? The only field that needs to be added in the child classes is the parent relation.

Reply:
#10697